src/tests/functional/rendering_tests.js

```diff
@@ -580,6 +582,15 @@ test("rendering a redirect response replaces the body once and only once", async
   assert.ok(await noNextBodyMutation(page), "replaces <body> element once")
 })
 
+
+test.only("hides progress bar when reloading due to request failure", async ({ page }) => {
+  await page.click("#download-link")
+  await page.waitForEvent('download')
+
+  await waitUntilNoSelector(page, ".turbo-progress-bar")
+  assert.notOk(await hasSelector(page, ".turbo-progress-bar"), "hides progress bar")
```

Comment:
If you rebase this branch, you'll be able to re-write this assertion using Playwright's `expect` assertions:

```suggestion
  await page.waitForEvent("download")

  await expect(page.locator(".turbo-progress-bar"), "hides progress bar").not.toBeAttached()
```